frontend/src/components/Modal.tsx

```diff
@@ -21,13 +21,13 @@ export function Modal({ isOpen, onClose, children }: ModalProps) {
   if (!isOpen) return null;
 
   return (
-    <div className="fixed inset-0 z-50 flex items-center justify-center p-4 sm:p-8">
+    <div className="fixed inset-0 z-50 flex items-center max-w-fit mx-auto justify-center p-3 sm:p-8">
       <div
         className="fixed inset-0 bg-black/50"
         onClick={onClose}
         aria-hidden="true"
       />
-      <div className="relative w-full max-w-4xl bg-white border-2 border-black shadow-sharp p-6 sm:p-8 overflow-y-auto max-h-[90vh] sm:rounded-lg">
+      <div className="relative w-full md:max-w-[425px] bg-white border-2 border-gray-300  p-6 sm:p-8 overflow-y-auto max-h-[90vh] sm:rounded-lg">
```

Comment:
Modal is thin, I don't think you max-w is applying because the content isn't flex-growing

Either grow or min width